### ISSUE TITLE
F-statistic nested fixed-effects OLS using clustered standard errors

### ISSUE THREAD

**Issue by Econ808** [bashtage/linearmodels#287]
Hi,

I am trying to compare two model specifications - one including extra interaction terms. I know the formula for conducting a basic F-test (Wooldridge 2013: Introductory econometrics):
![image](https://user-images.githubusercontent.com/36441764/96842303-2a523500-144d-11eb-8847-57f51333b158.png)
where SSRs are the restricted and unrestricted sum of squared residuals, respectively. Also:

![image](https://user-images.githubusercontent.com/36441764/96842795-c3814b80-144d-11eb-8c0f-077e5064bc5c.png)
![image](https://user-images.githubusercontent.com/36441764/96842816-caa85980-144d-11eb-8b9d-d599ad107463.png)

This test, however, is to my knowledge not robust, since the value is unchanged whether errors are cluster robust or not (same degrees of freedom and residual sum of squares). 

Basically it is a Waldtest where the null hypothesis is that all parameters added in the unrestricted model are 0. The R package https://www.rdocumentation.org/packages/lmtest/versions/0.9-38/topics/waldtest supposedly allows for this analysis, however, I've experienced that the plm package from R is painstakingly slow to the point where it is actually infeasible to estimate my model containing 34.000 entities and 6 million rows

Is there someway to perform this test such that it is robust?

**Comment by Econ808** [comment]
Basically it is a Waldtest where the null hypothesis is that all parameters added in the unrestricted model are 0. The R package https://www.rdocumentation.org/packages/lmtest/versions/0.9-38/topics/waldtest supposedly allows for this analysis, however, I've experienced that the plm package from R is painstakingly slow to the point where it is actually infeasible to estimate my model containing 34.000 entities and 6 million rows

**Comment by bashtage** [comment]
You can use the `wald_test` function on a results instance.

https://bashtage.github.io/linearmodels/panel/panel/linearmodels.panel.results.PanelResults.wald_test.html#linearmodels.panel.results.PanelResults.wald_test

F is not robust to clustering and heteroskedasticity.

**Comment by bashtage** [comment]
You need to estimate the model with the required cluster structure, and then you can use a Wald test.

**Comment by Econ808** [comment]
Amazing, thank you!
